### PR TITLE
`Plagiarism checks`: Fix empty clean up dialog

### LINKED ISSUE #8528: 'Plagiarism checks': UI bug
### Describe the bug

The clean-up button on plagiarism checks results in a bug in the UI (see the screenshot)

### To Reproduce

1. Run a plagiarism check
2. Clean up 
3. Verify you get the same box as in the screenshot 

### Expected behavior

Should not have this error in the UI

### Screenshots

image

### Which version of Artemis are you seeing the problem on?

7,0,2

### What browsers are you seeing the problem on?

Microsoft Edge

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by zhngharry**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context

This PR aims to fix #8528. When running a plagiarism check on a text-based exercise, there is no chart showing up + the clean up dialog is bugged. This is due to the sent data being faulty, and the client not handling it correctly.

### Description

The client is normally laid out to handle a plagiarismResult Object with a similarity distribution array of length 10. However, somehow the backend is sending an array of length 100 when checking for plagiarisms on Text-Based Exercises. This PR implements an algorithm that takes the bigger array and reduces the values inbetween (indexes Y0 - Y9) to one value and therefore transforms the array to length 10 to be passed to the ngx chart. By getting rid of the ngx chart instantiation errors, the modal is then able to be displayed properly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Text based Programming Exercise (no modeling exercise)

1. Create course
2. Create exercise
3. Submit two solutions as student
4. As instructor, run plagiarism check
5. Check whether the chart for similarity distribution is displayed properly and the "Clean Up" 

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Comment by iyannsch** [comment]
Thanks for your submission! I´m worried that we are currently fixing the symptom, not the cause of the problem - maybe a change in the server component in order to not return 100 quantizations but only 10 would make more sense? I think your comment in the code also hinted to that, correct?

I had a good look at the server code and got some pointers for you to follow along:
- The `JPlag` Module takes care of generating the plagiarism results as well as the similarity distribution.
- In the `de.jplag.JPlagResult.java` file, there is a configuration value set to 100 (line 27) even though the array is described to be of length 10 (line 24). 
- The API endpoints for Modeling-, Programming-, and TextPlagiarismChecks are quite similar when comparing them, you probably have to reverse-engineer it down to the library. 

I´m happy to postpone the server-changes to another PR if there are other components relying on the length of 100 in the result. Let´s discuss this! Maybe also other contributors with more knowledge in the server component can dig in here and add some reasoning @jakubriegel @pinarmrv @krusche

**Comment by zhngharry** [comment]
> Thanks for your submission! I´m worried that we are currently fixing the symptom, not the cause of the problem - maybe a change in the server component in order to not return 100 quantizations but only 10 would make more sense? I think your comment in the code also hinted to that, correct?
> 
> I had a good look at the server code and got some pointers for you to follow along:
> 
> * The `JPlag` Module takes care of generating the plagiarism results as well as the similarity distribution.
> * In the `de.jplag.JPlagResult.java` file, there is a configuration value set to 100 (line 27) even though the array is described to be of length 10 (line 24).
> * The API endpoints for Modeling-, Programming-, and TextPlagiarismChecks are quite similar when comparing them, you probably have to reverse-engineer it down to the library.
> 
> I´m happy to postpone the server-changes to another PR if there are other components relying on the length of 100 in the result. Let´s discuss this! Maybe also other contributors with more knowledge in the server component can dig in here and add some reasoning @jakubriegel @pinarmrv @krusche

Sounds good, and yes, because I'm not sure whether the server behavior was intended this way I commented in the code, that the changes right now should be reverted if the server's behavior is going to be changed. If the team agrees that the issue lies in the server, not the client, I try working on it. Thanks for the pointers!

**Comment by zhngharry** [comment]
@iyannsch Have there been any new updates regarding discussing this issue? I could try fixing this issue in the server

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-run-details/plagiarism-run-details.component.ts [hunk 1]
```diff
@@ -61,14 +61,44 @@ export class PlagiarismRunDetailsComponent extends PlagiarismAndTutorEffortDirec
     updateChartDataSet(data: number[]) {
         let ngxDataEntity;
         this.ngxData = [];
-        data.forEach((value, position) => {
+        this.totalDetectedPlagiarisms = 0;
+
+        // TextBased Exercises return 100 steps in data, Modeling only 10
```
Comment (krusche) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-run-details/plagiarism-run-details.component.ts:66]: I think we should fix this directly. Did you spot where this comes from?

**Review by krusche — CHANGES_REQUESTED** [review]
see my inline comment

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-run-details/plagiarism-run-details.component.ts (hunk 1 above)
Comment (zhngharry) [reply src/main/webapp/app/exercises/shared/plagiarism/plagiarism-run-details/plagiarism-run-details.component.ts]: Yes, I found it and fixed it. Locally it works now, but couldn't test it on the test servers myself

File: src/main/java/de/tum/in/www1/artemis/domain/plagiarism/text/TextPlagiarismResult.java [hunk 2]
```diff
@@ -29,7 +33,14 @@ public void convertJPlagResult(JPlagResult result, Exercise exercise) {
             this.comparisons.add(comparison);
         }
         this.duration = result.getDuration();
-        this.setSimilarityDistribution(result.getSimilarityDistribution());
+
+        // Convert JPlag Similarity Distribution from int[100] to int[10]
+        int[] tenthPercentileSimilarityDistribution = new int[REDUCED_SIZE];
+        for (int i = 0; i < ORIGINAL_SIZE; i++) {
+            tenthPercentileSimilarityDistribution[Math.floorDiv(i, REDUCED_SIZE)] += result.getSimilarityDistribution()[i];
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/domain/plagiarism/text/TextPlagiarismResult.java:40]:
```suggestion
            tenthPercentileSimilarityDistribution[i / REDUCED_SIZE] += result.getSimilarityDistribution()[i];
```
Normal integer division already floors the result

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by DominikRemo — APPROVED** [review]
Tested on ts4. LGTM

**Review by SimonEntholzer — APPROVED** [review]
Tested on TS4 in testing session, and works as expected :+1:

**Review by coolchock — APPROVED** [review]
tested during testing session, the clean up dialog was fixed and works as expected

**Review by EneaGore — APPROVED** [review]
Tested in TS4 in testing session. Works as expected.

**Review by BBesrour — APPROVED** [review]
code

**Review by kaancayli — APPROVED** [review]
Code LGTM 👍

**Review by iyannsch — APPROVED** [review]
Code LGTM

**Review by krusche — APPROVED** [review]

**Pull request merged**